ntoskrnl/ke/amd64/mp.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Architecture source file to hold multiprocessor functions
+ * COPYRIGHT:   Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES *****************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS *******************************************************************/
+
+#define PHYSICAL_ADDRESS  LARGE_INTEGER
+PHYSICAL_ADDRESS HighestPhysicalAddress;
+ULONG ProcessorCount = 0;
+
+struct APInfo
+{
+    KPCR pcr;
+    KTSS tss;
+    ETHREAD thread;
+    KDESCRIPTOR GdtDesc;
+    KDESCRIPTOR IdtDesc;
+} APInfo;
+
+/* FUNCTIONS *****************************************************************/
+VOID
+NTAPI
+KeStartAllProcessors()
+{
+    BOOLEAN ApplicationProcessorsEnabled = FALSE;
+    HighestPhysicalAddress.QuadPart = UINT64_MAX;
+    KDESCRIPTOR GdtDesc, IdtDesc;
+    __sgdt(&GdtDesc.Limit);
+    __sidt(&IdtDesc.Limit);
+    do
+    {
+        ProcessorCount++;
+        /* Attempt to allocate memory for its new setup */
+        SIZE_T APInfo = sizeof(struct APInfo) + 2; 
+        PVOID PAPInfo = MmAllocateContiguousMemory(APInfo, HighestPhysicalAddress);
+
+        if (!PAPInfo)
+        {
+            ASSERT("KeStartAllProcessors: Memory Allocation has failed");
+        }
```

Comment:
You're not using the ASSERT macro consistently. If you want to break out in the debugger if the predicate in the assertion condition is not TRUE (the predicate being your `PAPInfo` obviously) then use `ERROR_FATAL()`.
https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/reactos/debug.h;hb=fb8edf90a002cefd0d99adb4316f12e6d6840418#l238